mlflow/pyfunc/model.py

```diff
@@ -391,19 +391,330 @@ def predict_stream(
+                    "content": '{"content": "Successfully generated array of 5 random ints in [1,
+                    100].", "custom_outputs": {"random_nums": [93, 51, 12, 7, 25]}}',
+                    "name": "generate_random_ints",
+                    "id": "call_15ca4fcc-ffa1-419a-8748-3bea34b9c043",
+                    "tool_call_id": "call_15ca4fcc-ffa1-419a-8748-3bea34b9c043",
+                },
+                {
+                    "role": "assistant",
+                    "content": "The new set of generated random numbers are: 93, 51, 12, 7, and 25. Would you like me to generate more?",
+                    "name": "llm",
+                    "id": "run-70c7c738-739f-4ecd-ad18-0ae232df24e8-0",
+                },
+            ],
+            "custom_outputs": {"random_nums": [93, 51, 12, 7, 25]},
+        }
+
+    **Streaming Agent Output with ChatAgent**
+
+    The ChatAgent interface has a few streaming specific behaviors that are required by us and assumed by our helper LangChainChatAgent and LangGraphChatAgent classes.
+
+    - Streamed messages must be of the type :py:class:`ChatAgentChunk <mlflow.types.agent.ChatAgentChunk>`, with one message per chunk
+    - The ``custom_outputs`` key is meant to only be written to once during your agent's response. Old values will be overwritten in our LangGraphChatAgent example below.
+    - Messages from the same “node” of your ChatAgent should have the same ID. The content field of the message and usage stats of the :py:class:`ChatAgentChunk <mlflow.types.agent.ChatAgentChunk>` should be aggregated by the consuming client. See the example below.
+
+    .. code-block:: python
+
+        {"message": {"role": "assistant", "content": "Born", "id": "123"}}
+        {"message": {"role": "assistant", "content": " in", "id": "123"}}
+        {"message": {"role": "assistant", "content": " data", "id": "123"}}
+
+
+    **Authoring a ChatAgent**
+
+    Authoring an agent using the ChatAgent  interface is a framework-agnostic way to create a model with a  standardized interface that is loggable with the MLFlow pyfunc flavor, can be reused across clients, and is ready for serving workloads.
+
+    After creating an agent with any framework, you can wrap the framework's querying method with predict and predict_stream methods with the signature below that takes pydantic models. You can convert these Pydantic models back to dictionaries with `model_dump_compat` depending on what input format your framework of choice uses.
+
+    .. code-block:: python
+
+        def predict(
+            self,
+            messages: list[ChatAgentMessage],
+            context: Optional[ChatContext] = None,
+            custom_inputs: Optional[dict[str, Any]] = None,
+            ) -> ChatAgentResponse:
+
+    You can call the predict and predict_stream either with 3 pydantic models or a single dict that is of the :py:class:`ChatAgentRequest <mlflow.types.agent.ChatAgentRequest>` schema for ease of testing.
```

Comment:
The "3 pydantic models" piece is an implementation detail, can we emphasize the important takeaway which is that in addition to calling `predict` and `predict_stream` with inputs matching their type signatures, you can pass a dictionary to both methods - and it's good to link to an example (as it seems we're doing below) to show how that works